Framework/API/src/MatrixWorkspace.cpp

```diff
@@ -926,13 +924,21 @@ void MatrixWorkspace::setYUnitLabel(const std::string &newLabel) {
 * TODO: For example: ????
 * @return whether workspace is a distribution or not
 */
-const bool &MatrixWorkspace::isDistribution() const { return m_isDistribution; }
+bool MatrixWorkspace::isDistribution() const {
+  return getSpectrum(0).yMode() == HistogramData::Histogram::YMode::Frequencies;
```

Comment:
A `MatrixWorkspace` can never have zero spectra, the `initialize` method throws otherwise. Also many other things rely on this.